docs/common/widget.ldoc

```diff
@@ -39,3 +39,64 @@
 --- Set/get a widget's buttons.
 -- @param _buttons The table of buttons that should bind to the widget.
 -- @function buttons
+
+
+--- When the layout (size) change.
+-- @signal widget::layout_changed
```

Comment:
Could you add a comment that unless this signal is emitted, `:layout()` and `:fit()` must return the same result when called with the same arguments?
(And layout is more than just size, because it also involved `:layout()`. However, I don't have a good suggestion for the wording.)